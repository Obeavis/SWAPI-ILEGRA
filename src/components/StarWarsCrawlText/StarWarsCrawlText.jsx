import React from "react";
import PropTypes from "prop-types";
import Moment from 'react-moment';
import logo from "../../assets/images/star-wars-logo.png";
import "./StarWarsCrawlText.scss";

const StarWarsCrawlText = (props) => (
	<div className="star-wars-intro" key="aushauhsu">

		<p className="intro-text">
			A long time ago in a galaxy far, <br />
			far away...
        </p>

		<h2 className="main-logo">
			<img
				alt={props.film.title}
				src={logo}
			/>
		</h2>

		<div className="main-content">

			<div className="title-content">
				<p className="content-header">{`EPISODE ${props.film.episodeID}`} <br />
					{props.film.title}
				</p>
				<p className="content-body">
					{props.film.openingCrawl}
					<br /><br />
					<span>Direct by: {props.film.director}</span>
					<br /><br />
					<span>
						Release Date: <Moment format="LL">
							{new Date(props.film.releaseDate)}
						</Moment>
					</span>
				</p>

			</div>
		</div>
	</div>

);

StarWarsCrawlText.propTypes = {
	film: PropTypes.object
};

export default StarWarsCrawlText;