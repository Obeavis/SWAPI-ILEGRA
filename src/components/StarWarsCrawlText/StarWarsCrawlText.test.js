import React from "react";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import StarWarsCrawlText from "./StarWarsCrawlText";
import Moment from "react-moment";

configure({ adapter: new Adapter() });

describe("StarWarsCrawlText Component", () => {

    const film = {
        episodeID: 4,
        releaseDate: "1977-05-25",
        title: "A New Hope",
        director: "George Lucas",
        openingCrawl: "It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy...."
    };

    const component = (
        <StarWarsCrawlText film={film} />
    )

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Renders the image", () => {
        const wrapper = mount(component);
        const image = wrapper.find('img').prop("src");

        expect(image).toEqual('star-wars-logo.png');

    });

    it("Have the right title inside", () => {
        const wrapper = mount(component);
        const title = wrapper.find('.content-header').text();

        expect(title).toEqual('EPISODE 4 A New Hope');
    });

    it("Have the right text inside", () => {
        const wrapper = mount(component);
        const text = wrapper.find('.content-body').text();

        const date = mount(
            <Moment format="LL">
                {new Date(film.releaseDate)}
            </Moment>
        ).find('time').text();

        expect(JSON.stringify(text)).toEqual(JSON.stringify(`${film.openingCrawl}Direct by: ${film.director}Release Date: ${date}`));
    });

})