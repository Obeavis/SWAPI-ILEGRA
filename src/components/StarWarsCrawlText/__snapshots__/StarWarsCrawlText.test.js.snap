// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarWarsCrawlText Component matches the snapshot 1`] = `
<div
  className="star-wars-intro"
>
  <p
    className="intro-text"
  >
    A long time ago in a galaxy far, 
    <br />
    far away...
  </p>
  <h2
    className="main-logo"
  >
    <img
      alt="A New Hope"
      src="star-wars-logo.png"
    />
  </h2>
  <div
    className="main-content"
  >
    <div
      className="title-content"
    >
      <p
        className="content-header"
      >
        EPISODE 4
         
        <br />
        A New Hope
      </p>
      <p
        className="content-body"
      >
        It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....
        <br />
        <br />
        <span>
          Direct by: 
          George Lucas
        </span>
        <br />
        <br />
        <span>
          Release Date: 
          <time
            dateTime={"1977-05-25T00:00:00.000Z"}
          >
            May 24, 1977
          </time>
        </span>
      </p>
    </div>
  </div>
</div>
`;
